roles/branch-validation/tasks/main.yml

```diff
@@ -0,0 +1,205 @@
+---
+- name: create keypair
+  os_keypair:
+    auth:
+      username: admin
+      project_name: admin
+      password: "{{ secrets.admin_password }}"
+      auth_url: "http://127.0.0.1:{{ endpoints.keystone_admin.port.backend_api }}/{{ endpoints.keystone.version }}"
+    name: validate
+    state: present
+  register: keypair
+
+- name: persist private key
+  sudo: no
+  local_action:
```

Comment:
Also, as discussed, we do want to do this from the controller, it's not always the case that the API endpoints will be available to the bastion.